lib/Alien/Build/Plugin/Core/Setup.pm

```diff
@@ -83,6 +83,163 @@ sub _platform
+        $count = grep { /^cpu\d/ } <$fh>;
+        close $fh;
+      }
+      $ncpu = $count if $count;
+      last OS_CHECK;
+    };
+
+    /bsd|darwin|dragonfly/ && do {
+      chomp( my @output = `sysctl -n hw.ncpu 2>/dev/null` );
+      $ncpu = $output[0] if @output;
+      last OS_CHECK;
+    };
+
+    /aix/ && do {
+      my @output = `lparstat -i 2>/dev/null | grep "^Online Virtual CPUs"`;
+      if ( @output ) {
+        $output[0] =~ /(\d+)\n$/;
+        $ncpu = $1 if $1;
+      }
+      if ( !$ncpu ) {
+        @output = `pmcycles -m 2>/dev/null`;
+        if ( @output ) {
+          $ncpu = scalar @output;
+        } else {
+          @output = `lsdev -Cc processor -S Available 2>/dev/null`;
+          $ncpu = scalar @output if @output;
+        }
+      }
+      last OS_CHECK;
+    };
+
+    /gnu/ && do {
+      chomp( my @output = `nproc 2>/dev/null` );
+      $ncpu = $output[0] if @output;
+      last OS_CHECK;
+    };
+
+    /haiku/ && do {
+      my @output = `sysinfo -cpu 2>/dev/null | grep "^CPU #"`;
+      $ncpu = scalar @output if @output;
+      last OS_CHECK;
+    };
+
+    /hp-?ux/ && do {
+      my $count = grep { /^processor/ } `ioscan -fkC processor 2>/dev/null`;
+      $ncpu = $count if $count;
+      last OS_CHECK;
+    };
+
+    /irix/ && do {
+      my @out = grep { /\s+processors?$/i } `hinv -c processor 2>/dev/null`;
+      $ncpu = (split ' ', $out[0])[0] if @out;
+      last OS_CHECK;
+    };
+
+    /osf|solaris|sunos|svr5|sco/ && do {
+      if (-x '/usr/sbin/psrinfo') {
+        my $count = grep { /on-?line/ } `psrinfo 2>/dev/null`;
+        $ncpu = $count if $count;
+      }
+      else {
+        my @output = grep { /^NumCPU = \d+/ } `uname -X 2>/dev/null`;
+        $ncpu = (split ' ', $output[0])[2] if @output;
+      }
+      last OS_CHECK;
+    };
+
+    /mswin|mingw|msys|cygwin/ && do {
+      if (exists $ENV{NUMBER_OF_PROCESSORS}) {
+        $ncpu = $ENV{NUMBER_OF_PROCESSORS};
+      }
+      last OS_CHECK;
+    };
+
+    warn "CPU count: unknown operating system";
+  }
+
+  $ncpu = 1 if (!$ncpu || $ncpu < 1);
+
+  $ncpu;
+}
+
+sub _cpu_arch {
+  my ($my_config) = @_;
+
+  my $arch = {};
+
+  my %Config = %$my_config;
+
+  die "Config missing archname" unless exists $Config{archname};
+
+  if( $Config{archname} =~ m/
+      \b x64    \b # MSWin32-x64
+    | \b x86_64 \b # x86_64-linux
+    | \b amd64  \b # amd64-freebsd
+    /ix) {
+    $arch = { name => 'x86_64' };
+  } elsif( $Config{archname} =~ m/
+      \b x86  \b   # MSWin32-x86
+    | \b i386 \b   # freebsd-i386
+    | \b i486 \b   # i486-linux
+    | \b i686 \b   # i686-cygwin
+    /ix ) {
+    $arch = { name => 'x86' };
+  } elsif( $Config{archname} =~ m/
+      \b darwin \b
+    /ix ) {
+    my $cpu_brand = `sysctl -n machdep.cpu.brand_string`;
+    if( $cpu_brand =~ /Apple/ ) {
+      $arch = { name => 'aarch64' }; # Apple Silicon
+    } elsif( $cpu_brand =~ /Intel/ ) {
+      $arch = { name => 'x86_64' };  # Intel
```

Comment:
Probably totally unsupported by now, but Intel I think could technically be 32 bit x86.  I have hardware to check on so I can follow up on that.

Reply:
Oh, that would be great!

Possibly detected by either `$Config{ptrsize}` (could also be a 32-bit only Perl
running on a `x86_64` Apple and so it should return `x86` in case of linking)
or run one of these commands <https://apple.stackexchange.com/questions/12666/how-to-check-whether-my-intel-based-mac-is-32-bit-or-64-bit>.

Reply:
Interesting both of these are 10.4

32 bit Intel:
```
chaco% sysctl -n machdep.cpu.brand_string
Intel(R) Core(TM)2 CPU         T5600  @ 1.83GHz
```

PowerMac G5:
```
figment% sysctl -n machdep.cpu.brand_string
machdep: class is not implemented
```

Reply:
`sysctl -n hw.machine` might work better.  It appears to be:

 1. `Power Macintosh` on my PowerMac G5
 2. `i386` on my old 10.4 x86 mac mini
 3. `x86_64` on my iMac which is a little more recent
 4. `arm64` on M1 

`x86_64` and `Power Macintosh` won't technically give us the right word size since it could still be a 32 bit Perl as you pointed out, but we can use `$Config{ptrsize}`.